maven/bnd-resolver-maven-plugin/src/main/java/aQute/bnd/maven/resolver/plugin/ResolverMojo.java

```diff
@@ -69,6 +72,18 @@
 	@Parameter(defaultValue = "true")
 	private boolean						reportOptional;
 
+	@Parameter
+	@SuppressWarnings("serial")
+	private HashSet<Scope>				scopes	= new HashSet<Scope>() {
```

Comment:
What is this crazy code? Surely the parameter type should be `Set<Scope>`. And why do you need a HashSet anonymous subclass when you could simply use EnumSet.of?

Reply:
As I mentioned, EnumSet wont work. But neither will initialization with the default values since there would be no way to not have the default values in the set.

Reply:
Can you look at the updated code? Also, maven is really simple, It doesn't do ADDs to the preconfigured value, it just does a set, so if you have a default of:
```
	@Parameter
	private Set<Scope>					scopes	= EnumSet.of(Scope.compile, Scope.runtime);
```
and then the plugin is configured as:
```
<configuration>
	<scopes><scope>compile</scope></scopes>
</configuration>
```
it just does a `set` on the field with the assembled set. It does not try to rationalize the inputs against the defaults.

Reply:
it DOES however coerce the values into the member type. Which is where the validation will occur.

So
```
<configuration>
	<scopes><scope>conpile</scope></scopes>
</configuration>
```
will fail.